### PR TITLE
Migrate more howtos from MoinMoin

### PR DESCRIPTION
**Pull request by wohali**
In progress. So far:

* How to handle stats aggregation
* How to design for replication
* How to update views live without breaking production (a popular one!)
* Adding client-side security)

### CODE REVIEW

**Comment by wohali** [comment]
@flimzy ok, that's it for today, if you have time to review. If not I will continue to update this PR until you've done a pass. :)

File: src/best-practices/documents.rst [hunk 1]
```diff
@@ -48,3 +48,303 @@ to ensure unique identifiers for each row in a database table. CouchDB generates
...
+Designing an application to work with replication
+-------------------------------------------------
+
+Whilst CouchDB includes replication and a conflict-flagging mechanism, this is
+not the whole story for building an application which replicates in a way which
+users expect.
+
+Here we consider a simple example of a bookmarks application. The idea is that
+a user can replicate their own bookmarks, work with them on another machine,
+and then synchronise their changes later.
```
Comment (flimzy) [inline src/best-practices/documents.rst]: This reminds me of something I'd like to bring up again as a general discussion: Should we use US or UK English as the standard in this document? We mix a lot.  I don't have a strong opinion, except that I think the two most active proof-readers (@wohali and myself) are most familiar with US English. Is this worth bringing up on the ML?

File: src/best-practices/views.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+---------------------------------------------
+
+It is possible to change the definition of a view, build the index, then make
+those changes go live without causing downtime for your application. The trick
+to making this work is that CouchDB's JavaScript view index files are based on
+the contents of the design document - not its name, ``_id`` or revision. This
+means that two design documents with identical view code will share the same
+on-disk view index files.
+
+Here is a worked example, assuming your ``/db/_design/ddoc`` needs tobe updated.
```
Comment (flimzy) [inline src/best-practices/views.rst]:
```suggestion
Here is a working example, assuming your ``/db/_design/ddoc`` needs tobe updated.
```

Comment (wohali) [reply src/best-practices/views.rst]: https://www.merriam-webster.com/dictionary/worked

File: src/best-practices/documents.rst [hunk 3]
```diff
@@ -48,3 +48,303 @@ to ensure unique identifiers for each row in a database table. CouchDB generates
...
+periodically, and are disconnected for more than this time before they
+resynchronise.
+
+All of the approaches below which allow automated merging of changes rely on
+having some sort of history back in time to the point where the replicas
+diverged.
+
+CouchDB does not provide a mechanism for this itself. It stores arbitrary
+numbers of old _ids for one document (trunk now has a mechanism for pruning the
+_id history), for the purposes of replication. However it will not keep the
```
[the remark this answers is not in the document]
Comment (wohali) [reply src/best-practices/documents.rst]: Confirmed with @rnewson - `_revs_limit` only keeps around that number of `_rev` strings (of each leaf) for the purposes of ensuring consistency after a purge. Only leaf bodies are retained after compaction.

**Comment by wohali** [comment]
@flimzy Damn it, I committed my changes for the fixes you asked for above locally and forgot to push before merging. I'll just add those fixes to the next PR.

**Pull request merged**